### ISSUE TITLE
[bug report] plugin hierarchy messes up derive function in child plugin

### ISSUE THREAD

**Issue by eelkevdbos** [elysiajs/elysia#140]
I am using a plugin hierarchy for dependency injection, as described by the [docs](https://elysiajs.com/patterns/dependency-injection.html). However, I noticed that this does something weird with the derive method in "child" plugins.

The specific bug is replicated with the test below:
https://github.com/eelkevdbos/elysia/commit/5041a53504f5fbba1c2432d8d55bf2d058970685

The test inlined for clarity, comments added to this issue only:
```ts
const parent = new Elysia({ name: 'parent' }).derive(() => ({
	bye: () => 'bye'
}))

const child = new Elysia({ name: 'child' })
	.use(parent) // inlining this will make the test green
	.derive(({ bye }) => ({
		hi: () => `hi + ${bye()}`
	}))

const app = new Elysia()
	.use(parent) // inlining this will make the test green
	.use(child)
	.get('/', ({ hi }) => hi())

const res = await app.handle(req('/')).then((t) => t.text())
expect(res).toBe('hi + bye')
```

It will throw:
```TypeError: hi is not a function. (In 'hi()', 'hi' is undefined)```

**Comment by eelkevdbos** [comment]
Interestingly, the following succeeds (callable syntax):

```ts
const parent = (app: Elysia) =>
	app.derive(() => ({
		bye: () => 'bye'
	}))

const child = (app: Elysia) =>
	app.use(parent).derive(({ bye }) => ({
		hi: () => `hi + ${bye()}`
	}))

const app = new Elysia()
	.use(parent)
	.use(child)
	.get('/', ({ hi }) => hi())

const res = await app.handle(req('/')).then((t) => t.text())
expect(res).toBe('hi + bye')
```

**Comment by SaltyAom** [comment]
Fix with eabe264, available on 0.7.27.

Feel free to reopen if the issue still persists.
